server/src/tenant-manager/tenant-manager.service.ts

```diff
@@ -22,12 +24,21 @@ export class TenantManagerService {
     private readonly dataSourceService: DataSourceService,
   ) {}
 
+  private mainDataSource: DataSource;
+
   /**
    * Init a workspace by creating a new data source and running all migrations
-   * @param workspaceId
+   * @params workspaceId, prefillSeedData
    * @returns Promise<void>
    */
-  public async init(workspaceId: string): Promise<void> {
+  public async init(
+    workspaceId: string,
+    prefillSeedData = true,
+  ): Promise<void> {
+    if (await this.tenantDataSourceService.hasSchema(workspaceId)) {
```

Comment:
typeormService.createSchema already checks that I believe

Reply:
In which scenario we would call init on an existing tenant? If it's to make this method idempotent maybe we can check that where we call the service? I feel like in 99% of the case this extra query is unnecessary? What do you think?

Reply:
yes, you are right